python/gps/algorithm/cost/ioc_layers.py

```diff
@@ -50,9 +55,78 @@ def forward(self, bottom, top):
 
         # TODO(kevin) implement this (look at ioc_loss_layer.cpp in caffe)
         # python/gps/algorithm/policy_opt/policy_layers.py might also be helpful
+        loss = 0.0
+        dc = self.demo_counts_
+        sc = self.sample_counts_
+
+        # log importance weights of demos and samples.
+        d_log_iw = bottom[2].data
+        s_log_iw = bottom[3].data
+
+        # Sum over time and compute max value for safe logsum.
+        for i in xrange(self.num_demos):
+            dc[i] = 0.0
+            for t in xrange(self.T):
+                dc[i] += 0.5 * bottom[0].data[i, t, 0] # Not sure why the feature should be 0
```

Comment:
sorry, my comment above is wrong. bottom[0] is an Nd x T array containing the costs of the demo trajectories at each time step, and bottom[1] is an Ns x T array storing the costs of the samples at each time step.

So this can just be bottom[0].data[i, t]